### PR TITLE
Suggested changes to the Javascript Journeyman badge

### PR DESCRIPTION
**Pull request by hmsommer**
Suggested changes:

In the “Asynchronous Programming” micro badge:

- Removed “Make a `generators`, `yield` demo”.
   Reasoning: Generators aren’t asynchronous by default (the same way functions aren’t) and are already covered in the “ES2015 Features” micro badge. “Yield” isn’t specifically mentioned “ES2015 Features” but you can’t understand generators without also understanding the “yield” keyword, they come together the same way “switch” and “case” come together in switch statements.
- Added “Make an `then`, `catch`, `finally` promise demo (chaining)”.
   Reasoning: Hopefully everyone is just using async/await, but it’s still important to understand how the old stuff works since we maintain old code.

In the  “ES2015 Features” micro badge:

- Removed “Promises”.
   Reasoning: Promises were introduced as part of es6, however, there is a whole micro badge on asynchronous programming and so I’m removing to avoid duplication.
- Added the following items:
    - Object.fromEntries
    - Destructuring
    - Spread Operator, Rest Params
    - Enhanced Object Literals (syntax extension)
    - Symbols
    - Exponential Operator
    - Optional Catch binding
    - Nullish Coalescing Operator (??)
    - Optional Chaining Operator (?.)
    Reasoning: They were either part of es6 or have come out after es6 and I feel they are valuable to understand.
- Renamed the micro badge to “ES2015 Features and Beyond”.
   Reasoning: When this badge was originally written es6 was the shinny new and we weren’t thinking past that (I wasn’t anyway…). There are now new features that have come out since then and it would be nice to have a place to put those things. (Although as time goes on we might have to split those into their own micobadge. However most of the items added are es6 so there aren’t enough “beyond” items for it’s own badge at the moment)
- Removed the “ES6 Katas” option.
   Reasoning: The texts says to do all of the problems labeled as “easy” and says there should be at least 20. There are currently 78 marked as easy, plus it skips generators, iterators, rest, spread, and doesn’t include some of the newer items added to this mico badge.
- Added a link to “Exploring ES 7 & 8” in the study section.

In the “Functional JavaScript” micro badge:

- Added “Flat” and “FlatMap” to the Functional Javascript micro badge.
   Reasoning: These are useful functions that have come out since the time this was originally written.

### CODE REVIEW

File: badges-active/javascript/_micro_javascript-es2015-features.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
-# JavaScript - ES2015 Features
+# JavaScript - ES2015 Features and Beyond
```
Comment (gmdayley) [inline badges-active/javascript/_micro_javascript-es2015-features.md:1]: How about just renaming it to "Modern JavaScript"

File: badges-active/javascript/_micro_async-programming.md [hunk 2]
```diff
@@ -29,13 +29,13 @@ Finally, we will receive the gift of async/await to finally put this whole thing
 Choose one:
 
 1) Make a creative project that demonstrates all of the following:
-- [ ] Thorough use of Promises
-- [ ] Make a `generators`, `yeild` demo
+- [ ] Thorough use of Promises (including `all` and `allSettled`))
+- [ ] Make an `then`, `catch`, `finally` promise demo (chaining)
```
Comment (gmdayley) [inline badges-active/javascript/_micro_async-programming.md:33]:
```suggestion
- [ ] Make a `then`, `catch`, `finally` promise demo (chaining)
```

File: badges-active/javascript/_micro_async-programming.md [hunk 3]
```diff
@@ -29,13 +29,13 @@ Finally, we will receive the gift of async/await to finally put this whole thing
...
 - [ ] Make an `async`, `await` demo
 
 2) Demonstrate all of the following:
 - [ ] Complete all of either the nodeschool or Udacity course on promises
-- [ ] Make a `generators`, `yield` demo
+- [ ] Make an `then`, `catch`, `finally` promise demo (chaining)
```
Comment (gmdayley) [inline badges-active/javascript/_micro_async-programming.md:38]:
```suggestion
- [ ] Make a `then`, `catch`, `finally` promise demo (chaining)
```

**Review by gmdayley — APPROVED** [review]
Thank you Heidi.  This looks great.   

Do we address scoping and closures anywhere?

File: badges-active/javascript/_micro_javascript-es2015-features.md (hunk 1 above)
Comment (hmsommer) [reply badges-active/javascript/_micro_javascript-es2015-features.md:1]: Great idea! I will update that.

**Comment by hmsommer** [comment]
@gmdayley
Closures are in the "JavaScript Language Advanced" micro badge.
I changed `Let, Const` to `Let & Const (uses and scoping differences from var)` in the Modern JavaScript micro badge in hopes they will talk about block vs function scope.

**Pull request merged**